tests/integration/synapseclient/test_evaluations.py

```diff
@@ -188,12 +188,15 @@ def test_teams():
     team = syn.store(Team(name=name, description="A fake team for testing..."))
     schedule_for_cleanup(team)
 
-    found_team = syn.getTeam(team.id)
+    # not logged in, teams are public
+    anonymous_syn = Synapse()
```

Comment:
Is a "not signed in error" thrown when trying to use a function that requires authentication?

Reply:
This is pretty much why I marked this as a draft so far and not done yet. Right now you will get the error that the back end serves, which may not be user friendly as the current behavior of telling the user to log in.

I think the options are:
1. Just surface the back end error (as it is now).
2. On any 403 status code when the client is not logged in raise a SynapseAuthenticationError indicating you are not logged in (although I think there are endpoints which may not return a 403 even when the underlying cause is auth, for example the batch entity API I think will return a 201 always with individual failed responses, and it's also a bit specious because you might have tried to access something you don't have permission to even when logged in).
3. Keep a registry of known endpoints in the client that do not require auth, and raise an error before submitting the request if the endpoint is in that registry.

Let me know if you have suggestions per the above or others.

Reply:
FWIW I went with #2 of the above in subsequent commits.